src/utils/style-formatter.js

```diff
@@ -13,43 +19,10 @@ export default {
     const { font = {}, background = {}, border = {} } = style;
     // enable
     styles += disabled ? formatProperty("opacity", 0.4) : formatProperty("cursor", "pointer");
-    // font
-    styles += formatProperty("color", colorUtils.getColor(font, DEFAULTS.FONT_COLOR.color, theme));
-    const fontStyle = font.style || DEFAULTS.FONT_STYLE;
-    fontStyle.bold && (styles += formatProperty("font-weight", "bold"));
-    fontStyle.italic && (styles += formatProperty("font-style", "italic"));
-    // background
-    const primaryColor = theme.getDataColorSpecials().primary;
-    const backgroundColor = colorUtils.getColor(background, primaryColor, theme);
-    styles += formatProperty("background-color", backgroundColor);
-    // backgroundImage
-    if ((background.useImage || background.mode === "media") && background.url.qStaticContentUrl) {
-      let bgUrl = background.url.qStaticContentUrl.qUrl;
-      if (bgUrl) {
-        bgUrl = getImageUrl(bgUrl, app);
-        styles += formatProperty("background-image", `url('${bgUrl}')`);
-        styles += formatProperty("background-size", backgroundSize[background.size || DEFAULTS.BACKGROUND_SIZE]);
-        styles += formatProperty(
-          "background-position",
-          backgroundPosition[background.position || DEFAULTS.BACKGROUND_POSITION]
-        );
-        styles += formatProperty("background-repeat", "no-repeat");
-      }
-    }
-    // border
-    if (border.useBorder) {
-      const lengthShortSide = Math.min(element.offsetWidth, element.offsetHeight);
-      const borderColor = colorUtils.getColor(border, colorUtils.getFadedColor(backgroundColor), theme);
-      const borderSize = ((border.width || DEFAULTS.BORDER_WIDTH) * lengthShortSide) / 2;
-      styles += formatProperty("border", `${borderSize}px solid ${borderColor}`);
-      styles += formatProperty(
-        "border-radius",
-        `${((border.radius || DEFAULTS.BORDER_RADIUS) * lengthShortSide) / 2}px`
-      );
-    } else {
-      styles += "border: none;";
-    }
-
+    styles += getFontStyle(font, DEFAULTS.FONT_COLOR.color, theme, formatProperty);
+    styles += formatProperty("background-color", getBackgroundColor(background, theme));
+    styles += getBackgroundImage(background, app, formatProperty);
```

Comment:
This would append `undefined` to the `styles` string when `background` is falsy.